File: solutions/javascript/longest-words.js
module.exports = function (string) {
  var length = 0;
  // Store the list of longest words in an object to automatically filter
  // for duplicates.
  var hash   = {};

  // Simplistic splitting on spaces, could improve to trim punctuation as well.
  string.split(' ').forEach(function (word) {
    // If the word's length is longer than the previous longest, we want to
    // update the length and reset the hash back to be empty.
    if (word.length > length) {
      length = word.length;
      hash   = {};
    }

    // If the word length is the same as the current longest length, add the
    // lowercase version to the hash. Here, we could store the value as some
    // arbitrary value since it doesn't matter - if though we needed to return
    // the instance of the word we could store how we found the word as the
    // value and grab it out when we return.
    if (word.length === length) {
      return hash[word.toLowerCase()] = true;
    }
  });

  // Return an array with all the words.
  return Object.keys(hash);
};

File: solutions/ruby/longest_words.rb
def longest_words string
  # splitting the string into words and removing duplicates from the array
  words = string.downcase.split.uniq
  # finding the maximum value in the array based on length
  max = words.max_by { |word| word.length }
  # selecting all the words with the same length as the max value
  words = words.select { |word| word.length == max.length }
end